File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)

    answer = 0
    while len(scoville) >= 2:
        min = heapq.heappop(scoville)

        if min >= K:
            return answer

        min2 = heapq.heappop(scoville)

        heapq.heappush(scoville, min + (min2 * 2))
        answer += 1

    if heapq.heappop(scoville) >= K:
        return answer
    else:
        return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int n : scoville) {
            pq.offer(n);
        }
        
        int answer = 0;
        while (pq.peek() < K) {
            if (pq.size() > 1) {
                int s1 = pq.poll();
                int s2 = pq.poll() * 2;    
                pq.offer(s1 + s2);
            } else {
                if (pq.poll() < K) {
                    return -1;
                }
            }
            
            answer++;
        }
        
        return answer;
    }
}